### PR TITLE
Document MarkdownCheck plugin

### LINKED ISSUE #502: Can `linkchecker` be extendedd to Pandoc-flavoured markdown?
## Summary

Let `testfile.md` contain the following line:

`linkchecker` does not seem to work with Pandoc-flavoured Markdown links:

`<https://madeupname.com>` and `[My Website](https://madeupname.com)`

## Steps to reproduce

`linkchecker --check-extern testfile.md`

## Actual result

`That's it. 1 link in 1 URL checked. 0 warnings found. 0 errors found.`

## Expected result

St least one error depending on URL syntax that is recognized.

## Environment

lsb_release -a:
 * Operating system:  lsb_release -a: 
Distributor ID: ManjaroLinux
Description:    Manjaro Linux
Release:        20.1
Codename:       Mikah
 * Linkchecker version: 
 * Python version: LinkChecker 10.0.0.dev0 released xx.xx.xxxx
Copyright (C) 2000-2014 Bastian Kleineidam
 * Install method: Arch User Repo (AUR) for `linkchecker-git`
 * Site URL: 

## Logs

LinkChecker 10.0.0.dev0              Copyright (C) 2000-2014 Bastian Kleineidam
LinkChecker comes with ABSOLUTELY NO WARRANTY!
This is free software, and you are welcome to redistribute it
under certain conditions. Look at the file `LICENSE' within this
distribution.
Get the newest version at https://linkchecker.github.io/linkchecker/
Write comments and bugs to https://github.com/linkchecker/linkchecker/issues

Start checking at 2020-09-14 19:33:37+005

Statistics:
Downloaded: 0B.
Content types: 0 image, 0 text, 0 video, 0 audio, 1 application, 0 mail and 0 other.
URL lengths: min=23, max=23, avg=23.

That's it. 1 link in 1 URL checked. 0 warnings found. 0 errors found.
Stopped checking at 2020-09-14 19:33:39+005 (2 seconds)

## Other notes

Featrure request

### PR DESCRIPTION
**Pull request by cjmayo**
Closes #502.

### CODE REVIEW

File: doc/src/man/linkcheckerrc.rst [hunk 1]
```diff
@@ -503,6 +503,14 @@ WordParser
 Parse Word files for URLs to check. Needs the :pypi:`pywin32` Python
 extension installed.
 
+MarkdownCheck
+^^^^^^^^^^^^^
+
+Parse Markdown files for URLs to check.
+
+**filename_re=**\ *REGEX*
+    Regular expression matching the names of Markdown files.
```
Comment (mgedmin) [inline doc/src/man/linkcheckerrc.rst:512]: It would be nice to document the default value, or at least provide an example.

**Review by mgedmin — APPROVED** [review]
lgtm, but it could be even better! ;)

File: doc/src/man/linkcheckerrc.rst (hunk 1 above)
Comment (cjmayo) [reply doc/src/man/linkcheckerrc.rst:512]: I started with that (there are examples in other sections of the page), but then realised that none of the plugins have documented examples. Probably worth adding for all.

Comment (mgedmin) [reply doc/src/man/linkcheckerrc.rst:512]: A small improvement today is better than a large improvement next week/month/year/never.  ;)

**Review by anarcat — APPROVED** [review]

File: doc/src/man/linkcheckerrc.rst (hunk 1 above)
Comment (cjmayo) [reply doc/src/man/linkcheckerrc.rst:512]: Raised #506 to record.

**Pull request merged**